app/pages/sections/additional-services/order-item/controller.js

```diff
@@ -0,0 +1,187 @@
+  solutionName,
+  selectedPrice,
+  formData,
+}) => ({
+  ...selectedPrice,
+  serviceRecipient: {
+    name: serviceRecipientName,
+    odsCode: selectedRecipientId,
+  },
+  catalogueSolutionId: selectedSolutionId,
+  catalogueSolutionName: solutionName,
+  deliveryDate: extractDate('deliveryDate', formData),
+  quantity: parseInt(formData.quantity, 10),
+  estimationPeriod: formData.selectEstimationPeriod,
+  price: parseFloat(formData.price),
+});
+
+const postSolutionOrderItem = async ({
+  accessToken,
+  orderId,
+  selectedRecipientId,
+  serviceRecipientName,
+  selectedSolutionId,
+  solutionName,
+  selectedPrice,
+  formData,
+}) => {
+  const endpoint = getEndpoint({ api: 'ordapi', endpointLocator: 'postSolutionOrderItem', options: { orderId } });
+  const body = formatPostData({
+    selectedRecipientId,
+    serviceRecipientName,
+    selectedSolutionId,
+    solutionName,
+    selectedPrice,
+    formData,
+  });
+
+  await postData({
+    endpoint, body, accessToken, logger,
+  });
+  logger.info(`Order item for ${solutionName} and ${serviceRecipientName} successfully created for order id: ${orderId}`);
+  return { success: true };
+};
+
+const formatPutData = ({
+  formData,
+}) => ({
+  deliveryDate: extractDate('deliveryDate', formData),
+  quantity: parseInt(formData.quantity, 10),
+  estimationPeriod: formData.selectEstimationPeriod,
+  price: parseFloat(formData.price),
+});
+
+const putSolutionOrderItem = async ({
+  accessToken,
+  orderId,
+  orderItemId,
+  formData,
+}) => {
+  const endpoint = getEndpoint({ api: 'ordapi', endpointLocator: 'putCatalogueOrderItem', options: { orderId, orderItemId } });
+  const body = formatPutData({
+    formData,
+  });
+
+  await putData({
+    endpoint, body, accessToken, logger,
+  });
+  logger.info(`Order item successfully updated for order id: ${orderId} and order item id: ${orderItemId}`);
+  return { success: true };
+};
+
+export const saveSolutionOrderItem = async ({
+  orderId,
+  orderItemId,
+  accessToken,
+  selectedRecipientId,
+  serviceRecipientName,
+  selectedSolutionId,
+  solutionName,
+  selectedPrice,
+  formData,
+}) => {
+  try {
+    const response = orderItemId === 'neworderitem'
+      ? await postSolutionOrderItem({
+        accessToken,
+        orderId,
+        selectedRecipientId,
+        serviceRecipientName,
+        selectedSolutionId,
+        solutionName,
+        selectedPrice,
+        formData,
+      })
+      : await putSolutionOrderItem({
+        accessToken,
+        orderId,
+        orderItemId,
+        formData,
+      });
+    return response;
+  } catch (err) {
+    if (err.response.status === 400 && err.response.data && err.response.data.errors) {
+      logger.info(`Follow validation errors returned from the API ${JSON.stringify(err.response.data.errors)}`);
+      return err.response.data;
+    }
+    logger.error(`Error saving order item for ${solutionName} and ${serviceRecipientName} for order id: ${orderId}`);
```

Comment:
Could use catalogueItemName / catalogueItemId instead?

Reply:
yeah I will. Sorry not in this PR though